should minimum go version increase to version 1.17?

### Share your bug report, feature request, or comment.

I'll preface this first with this compiled perfectly fine on my Arch system with go1.19.2.

This issue I'm filing occurred with an Ubuntu 20.04 VPS using the minimum in the README of go1.16.

```
$ go version
go version go1.16 linux/amd64
$ cd owncast/
$ go run main.go
go: github.com/prometheus/procfs@v0.8.0 requires
	golang.org/x/sync@v0.0.0-20220601150217-0de741cfad7f: missing go.sum entry; to add it:
	go mod download golang.org/x/sync
```

I then ran `go mod download golang.org/x/sync`, which modified the go.sum file.

Then I tried running owncast again and got this error.

```
$ go run main.go
# golang.org/x/sys/unix
../.gvm/pkgsets/go1.16/global/pkg/mod/golang.org/x/sys@v0.0.0-20221010170243-090e33056c14/unix/syscall.go:83:16: undefined: unsafe.Slice
../.gvm/pkgsets/go1.16/global/pkg/mod/golang.org/x/sys@v0.0.0-20221010170243-090e33056c14/unix/syscall_linux.go:2255:9: undefined: unsafe.Slice
../.gvm/pkgsets/go1.16/global/pkg/mod/golang.org/x/sys@v0.0.0-20221010170243-090e33056c14/unix/syscall_unix.go:118:7: undefined: unsafe.Slice
../.gvm/pkgsets/go1.16/global/pkg/mod/golang.org/x/sys@v0.0.0-20221010170243-090e33056c14/unix/sysvshm_unix.go:33:7: undefined: unsafe.Slice
note: module requires Go 1.17
```

So I installed and set go1.17 to default. It was only then that Owncast started up and ran fine without any of the above errors. I was also able to get to its admin page.

--- a/README.md
+++ b/README.md
@@ -79,7 +79,7 @@ OBS, Streamlabs, Restream and many others have been used with Owncast. [Read mor
 1. Ensure you have pre-requisites installed.
     - C compiler, such as [GCC compiler](https://gcc.gnu.org/install/download.html) or a [Musl-compatible compiler](https://musl.libc.org/)
     - [ffmpeg](https://ffmpeg.org/download.html)
-1. Install the [Go toolchain](https://golang.org/dl/) (1.16 or above).
+1. Install the [Go toolchain](https://golang.org/dl/) (1.17 or above).
 1. Clone the repo. `git clone https://github.com/owncast/owncast`
 1. `go run main.go` will run from source.
 1. Visit `http://yourserver:8080` to access the web interface or `http://yourserver:8080/admin` to access the admin.

